applications/TreasuryTracker.md

```diff
@@ -0,0 +1,180 @@
+
+On-Chain Data Reading:
+Initial testing and development of functions for reading on-chain data have been performed. This represents a crucial step in achieving the project's goals, as it allows for accurate and real-time retrieval of referenda and treasury data directly from the blockchain.
+
+Research and Community outreach:
+Comprehensive research has been conducted to understand the needs of the community, the gaps in existing solutions, and the potential challenges in implementing the project.
+
+
+
+## Development Roadmap :nut_and_bolt:
+
+
+### Overview
+
+- **Total Estimated Duration:** 1 month
+- **Full-Time Equivalent (FTE):**  1.25
+- **Total Costs:** 10,000 USD
+
+### Milestone 1 Example � Basic functionality 
+
+> :exclamation: **The default deliverables 0a-0d below are mandatory for all milestones**, and deliverable 0e at least for the last one. 
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can use the application and its various features. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article** that explains what was done/achieved as part of the grant |
+| 1. | Backend Development | We will create a backend module that can accurately and efficiently read data from the blockchain  to extract relevant data and store it in a PostgreSQL database. The focus of this phase is to establish a robust data pipeline that ensures the integrity and reliability of the data used for analytics.|
+| 2. | Frontend Development | We will create a frontend user interface (UI) that will present the statistics defined in the Project details. This will involve the design and implementation of interactive charts and data tables to display data. The focus is on functionality with the minimum UI design.  |
+| 3. | Integration and Testing | The final milestone is focused on integrating the backend with the frontend, and conducting extensive testing to ensure everything functions as intended. The integration process will involve connecting the UI with the data stored in the database, ensuring smooth data flow and real-time updates. Testing will involve thorough checks for bugs, performance issues, and user experience improvements. This phase will finalize the platform and prepare it for public deployment.|
```

Comment:
> The integration process will involve connecting the UI with the data stored in the database

Are you're going to directly access the db from the UI? Or are you using the backend that you described earlier to do that? Or will it be a separate service?

Reply:
We will be using an ajax implementation and dynamically pulling the data for the charts. However we do intend to have page loads re-draw the page instead of using a react type package to separate all front end and api activity. There will likely be some pages that are drawn directly from the database in initial versions. We do not have a lot of experience with these react type packages and while it does seem pretty straight forward, for the purposes of getting this off the ground I would prefer to stick with what we know and refactor code later on to make it more "web3" later on. 

What seemed to be most critical at this point in time was getting the information necessary for decision making into as many hands as soon as possible.

Reply:
> There will likely be some pages that are drawn directly from the database in initial versions.

How are you going to authenticate these calls against the db server?

Reply:
This would all happen server side and the client/user would only see the compiled results. The calls would be made with standard "php-mysqlnd" package that connects to mysql via tcp+tls with user/pass authentication.

Reply:
Oh I see, so these db calls are made from some php-based backend service, right? I was confused because you made it sound like you're dispatching these requests from the frontend directly to the db (or at least I interpreted it this way) but it seems like that was a misconception. I found it confusing that the CakePHP MVC-Framework based service was labelled "Frontend" that connects to your other component that is also labelled with "Frontend".
Anyway, to clean up this confusion could you update your architecture diagram to reflect the deliverables? For example, I'm not sure if the "backend module that can accurately and efficiently read data from the blockchain to extract relevant data and store it in a PostgreSQL database" even exists on that diagram.

Reply:
I have updated the necessary documentation and architecture image